gridftp/server/src/globus_gridftp_server.c

```diff
@@ -613,7 +612,8 @@ globus_l_gfs_ipc_closed(
     }
 
     handle = (globus_xio_handle_t) user_arg;
-    globus_mutex_unlock(&globus_l_gfs_mutex);
+
+    globus_mutex_lock(&globus_l_gfs_mutex);
```

Comment:
The original code was definitely wrong here, but why did that not create an issue? Maybe `globus_mutex_unlock()` just returns if the mutex wasn't locked.

Reply:
Unlocking a normal pthread mutex the calling thread has not locked is undefined behaviour, so the implementation can do whatever. See the pthread_mutex_lock man page. It usually just does nothing, but it is not guaranteed.

Reply:
Ok.